### PR TITLE
[ResourceDetectors.Azure] update changelog and readme

### PR DESCRIPTION
**Pull request by TimothyMothra**
## Changes
- update changelog
- update readme for VM detector.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.ResourceDetectors.Azure/README.md [hunk 1]
```diff
@@ -31,3 +31,53 @@ var tracerProvider = Sdk.CreateTracerProviderBuilder()
                             .AddDetector(new AppServiceResourceDetector()))
                         .Build();
 ```
+
+The resource detector will record the following attributes:
+
+|                         |
```
Comment (utpilla) [inline src/OpenTelemetry.ResourceDetectors.Azure/README.md:37]: nit:

This renders a table with an empty header.

We could probably get rid of the line above which says: "The resource detector will record the following attributes:" and add some text as the table header instead. Something like "Attributes recorded by AppServiceResourceDetector".

Comment (TimothyMothra) [reply src/OpenTelemetry.ResourceDetectors.Azure/README.md]: Done, thanks!
still going back-and-forth with Vishwesh to decide the best way to display this information :)

File: src/OpenTelemetry.ResourceDetectors.Azure/README.md [hunk 2]
```diff
@@ -31,3 +31,50 @@ var tracerProvider = Sdk.CreateTracerProviderBuilder()
...
+Adds resource attributes for the applications running in an Azure virtual machine.
+The following example shows how to add `AzureVMResourceDetector` to
+`TracerProvider` configuration:
+
+```csharp
+using OpenTelemetry;
+using OpenTelemetry.ResourceDetectors.Azure;
+using OpenTelemetry.Resources;
+
+var tracerProvider = Sdk.CreateTracerProviderBuilder()
```
Comment (utpilla) [inline src/OpenTelemetry.ResourceDetectors.Azure/README.md:59]: It would be good to also add the code snippet showing how to add these for metrics.

**Review by vishweshbankwar — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

File: src/OpenTelemetry.ResourceDetectors.Azure/README.md (hunk 2 above)
Comment (TimothyMothra) [reply src/OpenTelemetry.ResourceDetectors.Azure/README.md:58]: Why only Metrics? The Metrics API is practically identical to the Trace API.
Should we also add a Log example?

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.ResourceDetectors.Azure/README.md [hunk 3]
```diff
@@ -31,3 +32,50 @@ var tracerProvider = Sdk.CreateTracerProviderBuilder()
...
+as well.
+
+```csharp
+using OpenTelemetry;
+using OpenTelemetry.ResourceDetectors.Azure;
+using OpenTelemetry.Resources;
+
+var tracerProvider = Sdk.CreateTracerProviderBuilder()
+                        // other configurations
+                        .SetResourceBuilder(ResourceBuilder
```
Comment (utpilla) [inline src/OpenTelemetry.ResourceDetectors.Azure/README.md:62]: Use `ConfigureResource` instead.

`.ConfigureResource(resource=> resource.AddDetector(new AzureVMResourceDetector()))`

Comment (utpilla) [reply src/OpenTelemetry.ResourceDetectors.Azure/README.md:62]: @TimothyMothra Something to be considered for a follow-up PR.

**Pull request merged**